src/index.js

```diff
@@ -1,75 +1,112 @@
-export { mix, is, like }
+'use strict';
 
-function mix(...args) {
-	var superclass = !is(args[0], 'factory') && args.shift() || baseclass
-	var factory = (is(args[args.length-1], 'factory') && args.pop()) || derive
-	superclass = is(superclass, 'mixin') ? superclass.class : derive(superclass)
-	if (args.length) factory = (org => superclass => org(args.reduce((s,m) => m.mixin(s), superclass)))(factory)
-	function mixin(superclass) {
-		var result = is(superclass, mixin) ? superclass : factory(superclass) 
-		if (mixin.classes.indexOf(result) === -1) mixin.classes.push(result)
-		return result
-	}
-	Object.defineProperties(mixin, {
-		classes: {value:[], writable:false},
-		mixins: {value:args, writable:false},
-	})
-	var Class = mixin(superclass)
-	var constructor = Class.hasOwnProperty('constructor') ? Class.constructor.bind(Class) : (...args) => new Class(...args)
-	Object.getOwnPropertyNames(Class).forEach(k => Object.defineProperty(constructor, k, {value: Class[k]}))
-	return Object.defineProperties(constructor, {
-		mixin: {value:mixin, writable:false},
-		class: {value: Class, writable:false},
-		interface: {get:(x => () => x ? x : x = getInterface(Class.prototype))()},
-	})
-}
+const baseclass = class Object {};
+const derive = superclass => ({}[superclass.name || 'Object'] = class extends superclass {});
+
+const getPropertyNames = function getPropertyNames(proto) {
```

Comment:
Been reading the AirBnB guide but afaict their argument is just 'hoisting is bad'.
Personally I feel hoisting is a feature that is not that hard to understand and actually has some advantages as well. But hey I also love the ternary operator `?` and the ugly `|| defaultValue` trick. I like shortcuts. Because they are short ;) 

My problem with linting is that it ends up as a religious debate.

Ask yourself: why did AirBnB create their own coding standard? Are they really the first to do this? Why didn't they adopt e.g. https://standardjs.com/ or any of the dozens of other 'standards' in use?